src/dash/utils/TimelineConverter.js

```diff
@@ -150,7 +150,11 @@ function TimelineConverter() {
         const now = calcPresentationTimeFromWallTime(new Date(), voPeriod);
         const periodEnd = voPeriod.start + voPeriod.duration;
         range.start = Math.max((now - voPeriod.mpd.timeShiftBufferDepth), voPeriod.start);
-        range.end = now >= periodEnd && now - d < periodEnd ? periodEnd - d : now - d;
+        if (lowLatencyMode) {
+            range.end = now >= periodEnd && now < periodEnd ? periodEnd : now;
```

Comment:
This doesn't look right - the first chunk isn't available until after the chunk duration. Just subjecting the segment duration could lead to segments being requested early.

Reply:
@bbcrddave, that seems to be true for regular streams but for low latency ones a chunk should start being available as soon as the duration of its first sub-chunk. Furthermore, if HTTP server supports chunked transfer encoding, and the process in charge of generating chunks is flushing each sub-chunk before they are fully completed, we could start downloading the chunk as soon as the server has something ready to be served (first frames encoded?).

Our above suggestion looks a bit aggressive (we assume "now" is covered by the latest segment), but this is just to inform about segments availability. We are assuming segment for live time is available (it should almost be), although later on, when calculating edge live position, we apply a safety margin that protects player for requesting a time position that still is not covered in the latest segment. 

Unfortunately there is nothing in the mpd (or at least nothing that I know) that indicates the duration of sub-chunks so the value of the safety margin used for calculating edge live position is something we should expose though Dash.js api. When trying to reduce latency as much as possible, it is not the same having sub-chunks of 1 second, than sub-chunks of 0.5 seconds.

Reply:
Unless the mdat is being sent before the moof is written (in which case the player would need to be of this and reorder the boxes before they are appended to be compliant with MSE), a CMAF chunk should only be available once the moof has been written. By definition this cannot occur until all the mdat for that chunk have been completed, so that would be the earliest point at which anything would be available, regardless of the transfer mode,

We shouldn't be assuming anything regarding availability - the client should know what is available at all times. The lowest latency would be achieved by requesting chunks/segments as soon as they are available.

In terms of manifest signalling, whilst I am not sure that it was the original intention, it seems that some streams intend to use `@availabilityTimeOffset` in order to signal the offset to the availabilityStartTime at which the first chunk becomes available. I think `@availabilityTimeComplete` should probably be set to `false` to indicate the segment is not complete at the offset time, but I have not seen that. At least if these or some similar signalling were used, clients not able to understand the signalling would not be affected.

It seems to me that there really needs to be some standardisation around this topic before much more work is done, as currently there seems to be no authoritative test streams or player requirements, or content production guidelines for that matter. Some work is due to kick off in DVB this week. Not sure if DASH-IF has anything planned (or has already completed something and I have not seen it).

Reply:
You are right, the minimum unit would be a CMAF chunk (as you said, moof atom can only be built, and written, once we have built the mdat one), independently of using chunked transfer encoding or not.

@bbcrddave, diving into some of the latest published information regarding DASH and low latency, seems people agree on using @availabilityTimeOffset to signal availability of the first chunk within a segment and @availabilityTimeComplete sets to false. Looking at DASH-IF IOP 4.1, its description for @availabilityTimeOffset confirms that:

> If files are available in chunks on the origin, for example due to specific encoding or delivery matters, chunked delivery may be supported. If this feature is offered, then the @availabilityTimeOffset attribute may be provided to announce how much earlier than the nominal segment availability the segment can be accessed. 

That also matches with your comment.

According to this, we are going to modify timing behavior so we take into account availabilityTimeOffset instead of using an arbitrary value.

Reply:
@epiclabsDASH  - this interpretation and use of @availabilitytimeoffset with chunked low latency encoding is correct. If a stream has 1s segments with 100ms chunks, then @availabilitytimeoffset should be set to 0.9. This instructs the player that it can make a request for that segment 0.9s earlier than it would if the content were non-chunked.

Reply:
Great. I am going to push a commit in the next few minutes with that interpretation implemented. It is now under testing. Thanks to both